### PR TITLE
Feature/6089 configure error correction level

### PR DESCRIPTION
**Pull request by naveeddotio**
## Description
According to tech specs:

**Error Correction Level: the error correction mode shall be set according to the value of Configuration Parameter .qrCodeErrorCorrectionLevel.**

Error correction level was missing in the current QR code generation, this PR is about fixing this issue.

## Screenshots
NA

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift [hunk 1]
```diff
@@ -9,17 +9,17 @@ extension TraceLocation {
 
 	// MARK: - Internal
 	
-	func qrCode(size: CGSize = CGSize(width: 400, height: 400)) -> UIImage? {
+	func qrCode(size: CGSize = CGSize(width: 400, height: 400), qrCodeErrorCorrectionLevel: String = "M") -> UIImage? {
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift:12]: Can you Please use an enum for this? qrCodeErrorCorrectionLevel is already an enum inside the protobuf we should keep it like this, if you need the string maybe create a property inside the enum to return the corrosponding letter but i think an enum would be enough

File: src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelProvider.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	private var subscriptions: Set<AnyCancellable> = []
+
+	func errorCorrectionLevel(
+		appConfigurationProvider: AppConfigurationProviding,
+		onCompletion: @escaping((String) -> Void)
+	) {
+		appConfigurationProvider.appConfiguration().sink { appConfig in
+			let qrCodeErrorCorrectionLevel = appConfig.presenceTracingParameters.qrCodeErrorCorrectionLevel
+			var mappedErrorCorrectionLevel = "M"
+
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelProvider.swift:25]: this is the same regarding the enum maybe here is a good place to do so.
`var mappedErrorCorrectionLevel: QRCodeErrorCorrectionLevel = .medium`

**Review by 30mar — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift (hunk 1 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift:12]: Yes, we need string letters; already present enums are meaningless for QR code generation, adding a property with in the enum of pb files will already change them, have to find an alternative.

File: src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelProvider.swift (hunk 2 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelProvider.swift:25]: It doesn’t make sense, we need letters and not numbers for QR code generation.

File: src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift (hunk 1 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift:12]: you can declare enums with string property and use the cases as string or at least their raw values

File: src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelProvider.swift (hunk 2 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelProvider.swift:25]: you can declare enums with string property and use the cases as string or at least their raw values

Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelProvider.swift:25]: or you could pass it as associated value in the enum case

File: src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift (hunk 1 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Extensions/TraceLocation+GenerateQRCode.swift:12]: or you could pass it as associated value in the enum case

File: src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelType.swift [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import Foundation
+
+enum MappedErrorCorrectionType: Int {
+	case medium
+	case large
+	case quantile
+	case high
+	
+	var mappedValue: String {
+		switch self {
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Services/Event Registration/QRCodePosterTemplate/QRCodeErrorCorrectionLevelType.swift:14]: nice

**Review by 30mar — APPROVED** [review]

**Review by pascal-brause — APPROVED** [review]
So much better :)

**Pull request merged**